numpy/ma/tests/test_core.py

```diff
@@ -1990,6 +1997,226 @@ def test_datafriendly_mul_arrays(self):
         assert_equal(a, [[1, 1], [3, 3]])
         assert_equal(a.mask, [[0, 1], [0, 1]])
 
+    def test_inplace_addition_scalar_type(self):
+        # Test of inplace additions
+        for t in self.othertypes:
+            with warnings.catch_warnings(record=True) as w:
+                warnings.filterwarnings("always")
+                (x, y, xm) = (_.astype(t) for _ in self.uint8data)
+                xm[2] = masked
+                x += t(1)
```

Comment:
Yeah, so I had snuck in this line above [1727](https://github.com/numpy/numpy/pull/5563/files#diff-14bc57103bf08898d2d8008d3baa765eR1727). I added a line note in the diff just in case it got lost. This should make sure they are types. Unless, I am not understanding what you are meaning.